repo2docker/buildpacks/julia/julia_project.py

```diff
@@ -63,7 +63,8 @@ def get_build_env(self):
             ('JULIA_BINDIR', '${JULIA_PATH}/bin'),
             ('JULIA_DEPOT_PATH', '${JULIA_PATH}/pkg'),
             ('JULIA_VERSION', self.julia_version),
-            ('JUPYTER', '${NB_PYTHON_PREFIX}/bin/jupyter')
+            ('JUPYTER', '${NB_PYTHON_PREFIX}/bin/jupyter'),
+            ('JUPYTER_DATA_DIR', '${APP_BASE}/conda/share/jupyter')
```

Comment:
Is this guaranteed to be the path to the kernels? Is it conceivable that some configurations could have it elsewhere?

Reply:
@betatim do you know?

Reply:
Since no one is chiming in here, I guess it is ok as is?

Reply:
Why do we set this new variable in the first place?

Reply:
`NB_PYTHON_PREFIX` points to the conda env&python executable which is used to start the jupyter notebook server. This is always Python 3. In some cases this is the same environment as the environment setup for the user/for the repository.

There is also `KERNEL_PYTHON_PREFIX` which points to the conda environment in which the Python executable for the user's environment is.

They are different in things like https://mybinder.org/v2/gh/binder-examples/python2_runtime/master

This is what it looks like for that binder:
```
jovyan@jupyter-binder-2dexamples-2dpython2-5fruntime-2dtoi64xwr:~$ jupyter kernelspec list
Available kernels:
  python2    /srv/conda/share/jupyter/kernels/python2
  python3    /srv/conda/share/jupyter/kernels/python3
jovyan@jupyter-binder-2dexamples-2dpython2-5fruntime-2dtoi64xwr:~$ echo $NB_PYTHON_PREFIX
/srv/conda
jovyan@jupyter-binder-2dexamples-2dpython2-5fruntime-2dtoi64xwr:~$ echo $KERNEL_PYTHON_PREFIX
/srv/conda/envs/kernel
```

If you look inside `/srv/conda/share/jupyter/kernels/python2/kernel.json` it points to the Python executable in `/srv/conda/envs/kernel`.

Overall I think we should mirror this.
```suggestion
            ('JUPYTER_DATA_DIR', '${NB_PYTHON_PREFIX}/share/jupyter'),
```

---

(We shouldn't treat "no response" as consent, it is more likely to mean that people are doing other things. Especially threads with discussion/arguments/lots of traffic (it tends to correlate with disagreement/controversy) are the ones that I personally de-prioritise first when crunched for time. Not saying this is one of those, just as a general point. Busy threads can easily consume 20-30min of thinking time which means I can tackle maybe one of these per working day. On the contrary PRs I prioritise are those that clearly layout the options/research done into options and explain why they decided to do things the way they did. This allows me to read the argument and say yes to it as it answers the two big questions I have during review: why make this change and why make this change in this way.)

Reply:
@betatim - I apologize for my unnecessary comment. Clearly "no response" is not consent. I understand completely that the maintainers are busy and can't be expected to respond immediately to every mention. This PR is blocking some downstream progress for me, so I am eager to see it resolved, but that is no justification for trying to rush the process.

FWIW, in #621 I tried to be more verbose about the need for this change, but that PR was closed in favor of this one.

Thanks for your valuable feedback.

Reply:
No worries.

I think the long term solution is to find a way for more people to join and feel able to review and merge work. Otherwise we are one or two buses away from disaster :)